docs/import/add_disaster.js

```diff
@@ -60,14 +76,235 @@ function enableWhenReady() {
+          case STATE.QUEUED_WRITE:
+            loadingElementFinished(writeWaiterId);
+            return updateLayersInFirestore();
+          case STATE.SAVED:
+            console.error('Unexpected layer write state');
+            return null;
+        }
+      });
+}
+
+/**
+ * Update the table and disasterData with new indices after a layer has been
+ * reordered. Then write to firestore.
+ * @param {Object} ui jquery object that contains details about this sort
+ * @return {?Promise<void>} See updateLayersInFirestore doc
+ */
+function updateAfterSort(ui) {
+  const layers = getCurrentLayers();
+  const numLayers = layers.length;
+  const oldRealIndex = $(ui.item).children('.index-td').html();
+  const newRealIndex = numLayers - 1 - $(ui.item).index();
+
+  // pull out moved row and shuffle everything else down
+  const row = layers.splice(oldRealIndex, 1)[0];
+  // insert at new index
+  layers.splice(newRealIndex, 0, row);
+
+  // dumbly renumber all the layers. note: nth-child starts at 1, .index()
+  // starts at 0
+  for (let i = 1; i <= numLayers; i++) {
+    $('#tbody tr:nth-child(' + i + ') .index-td').html(numLayers - i);
+  }
+
+  return updateLayersInFirestore();
+}
+
+/**
+ * Utility method for creating table divs with a given inner text
+ * @return {JQuery<HTMLTableDataCellElement>}
+ */
+function createTd() {
+  return $(document.createElement('td'));
+}
+
+/**
+ * Add text to the given td.
+ * @param {JQuery<HTMLTableDataCellElement>} td cell
+ * @param {Object} layer
+ * @param {string} property
+ * @return {JQuery<HTMLTableDataCellElement>}
+ */
+function withText(td, layer, property) {
+  return td.html(layer[property]);
+}
+
+/**
+ * Add a sample of info from a list to the given td.
+ * @param {JQuery<HTMLTableDataCellElement>} td cell
+ * @param {Object} layer
+ * @param {string} property
+ * @return {JQuery<HTMLTableDataCellElement>}
+ */
+function withList(td, layer, property) {
+  return td.html(layer[property][0]);
+}
+
+const layerTypeStrings = new Map();
+for (const t in LayerType) {
+  if (LayerType.hasOwnProperty(t)) {
+    layerTypeStrings.set(LayerType[t], t);
+  }
+}
+
+/**
+ * Add layer type info to the given td.
+ * @param {JQuery<HTMLTableDataCellElement>} td cell
+ * @param {Object} layer
+ * @param {string} property
+ * @return {JQuery<HTMLTableDataCellElement>}
+}
+ */
+function withType(td, layer, property) {
+  return td.html(layerTypeStrings.get((layer[property])));
+}
+
+/**
+ * Add checkbox capabilities to the given td.
+ * @param {JQuery<HTMLTableDataCellElement>} td cell
+ * @param {Object} layer
+ * @param {string} property
+ * @return {JQuery<HTMLElement> }
+ */
+function withCheckbox(td, layer, property) {
+  const checkbox = $(document.createElement('input'))
+                       .prop('type', 'checkbox')
+                       .prop('checked', layer[property])
+                       .on('change', (event) => {
+                         const index = $('#tbody > tr').length -
```

Comment:
This seems like a pretty indirect way to find the index (and it also seems expensive, since I'm guessing the $('tr').index below has to do at least a linear search).

Why not use the index-td of this row directly?